Source/WebCore/accessibility/AXObjectCache.cpp

```diff
@@ -3708,11 +3708,17 @@ std::optional<TextMarkerData> AXObjectCache::textMarkerDataForVisiblePosition(co
 
         unsigned differenceBetweenDomAndRenderedOffsets = textBox->minimumCaretOffset();
         unsigned previousEndDomOffset = textBox->maximumCaretOffset();
+        size_t previousLineIndex = textBox->lineIndex();
 
         while (domOffset > textBox->maximumCaretOffset()) {
             textBox = InlineIterator::nextTextBoxInLogicalOrder(textBox, orderCache);
-            differenceBetweenDomAndRenderedOffsets += textBox->minimumCaretOffset() - previousEndDomOffset;
+            size_t newLineIndex = textBox->lineIndex();
+            unsigned differenceToPrevious = textBox->minimumCaretOffset() - previousEndDomOffset;
+            // Just like when building AXTextRuns, we need to consider trimmed spaces between lines. So, if we find a gap between runs, subtract one from that gap to account for a trimmed space.
```

Comment:
This comment is long, might display in editors a bit nicer if we split it into two lines?